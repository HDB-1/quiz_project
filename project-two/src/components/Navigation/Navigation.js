import React from 'react'

export default function Navigation(props) {
    return (
        <div>
            <button id="nextBtn" onClick={props.nextQuestion}></button>
            <button id="previousBtn" onClick={props.previousQuestion}></button>
        </div>
    )
}
