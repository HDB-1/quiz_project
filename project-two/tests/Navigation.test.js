import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../src/components/Navigation/Navigation'

describe('Shallow Navigation', () => {
  let wrapper;
  let nextQuestionMock = jest.fn()
  let previousQuestionMock = jest.fn()
  beforeEach(() => wrapper = shallow(<Navigation nextQuestion={nextQuestionMock} previousQuestion={previousQuestionMock}/>)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render Next Question and Previous Question buttons',()=>{
      expect(wrapper.find('button').length).toEqual(2)
      expect(wrapper.find('#nextBtn').length).toEqual(1)
      expect(wrapper.find('#previousBtn').length).toEqual(1)
  })
  it('Next button should call nextQuestion function', () => {
      wrapper.find('#nextBtn').simulate('click')
      expect(nextQuestionMock).toHaveBeenCalled()
  });
  it('Previous button should call previousQuestion function', () => {
    wrapper.find('#previousBtn').simulate('click')
    expect(previousQuestionMock).toHaveBeenCalled()
  });
});