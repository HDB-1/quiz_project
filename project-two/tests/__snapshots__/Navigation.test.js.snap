// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Navigation Should render correctly 1`] = `
<div>
  <button
    id="nextBtn"
    onClick={[MockFunction]}
  >
    Next Question
  </button>
  <button
    id="previousBtn"
    onClick={[MockFunction]}
  >
    Previous Question
  </button>
</div>
`;
